### PR TITLE
Fix Exception When Opening Context Menu on Scripted Plots

### LINKED ISSUE #28187: Exception when you right-click a non-workspace plot
```
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot([0,1],[0,1])
fig.show()
```

Right-click the figure.

```
Traceback (most recent call last):
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\cbook\__init__.py", line 216, in process
    func(*args, **kwargs)
  File "C:\MantidNightlyInstall\bin\workbench\plotting\figureinteraction.py", line 138, in on_mouse_button_press
    self._show_context_menu(event)
  File "C:\MantidNightlyInstall\bin\workbench\plotting\figureinteraction.py", line 299, in _show_context_menu
    self._add_normalization_option_menu(menu, event.inaxes)
  File "C:\MantidNightlyInstall\bin\workbench\plotting\figureinteraction.py", line 326, in _add_normalization_option_menu
    can_toggle_normalization = self._can_toggle_normalization(ax)
  File "C:\MantidNightlyInstall\bin\workbench\plotting\figureinteraction.py", line 709, in _can_toggle_normalization
    axis = ax.creation_args[0].get('axis', None)
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**

**To test:**
1. Open workbench
2. Run this script:
 ``` python
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot([1,2],[1,2])
fig.show()
 ```
3. Right click the plot
4. No normalisation options should appear and there should be no exception raised. 

Fixes #28187 

*This does not require release notes* because **it fixes a bug introduced this release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
Should this be for merging into master or release next?

**Comment by ConorMFinn** [comment]
Release-next. Thanks, forgot about that bit.

**Review by Harrietbrown — DISMISSED** [review]
The context menu opens without error and no normalization option is enabled for the line, no errors are found with overplotting workspaces onto the plot. Good Job.

File: qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py [hunk 1]
```diff
@@ -314,6 +314,14 @@ def test_scripted_plot_show_and_hide_all(self):
         self.interactor.errors_manager.toggle_all_errors(self.ax, make_visible=True)
         self.assertTrue(self.ax.containers[0][2][0].get_visible())
 
+    def test_no_normalisation_options_on_scripted_plot(self):
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py]: This is a nice test. :+1: Can I be uber picky and ask that we rename it to something like `test_no_normalisation_options_on_non_workspace_plot`? A scripted plot from a workspace should be okay, I think, just not one directly from arrays?

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py [hunk 2]
```diff
@@ -314,6 +314,13 @@ def test_scripted_plot_show_and_hide_all(self):
         self.interactor.errors_manager.toggle_all_errors(self.ax, make_visible=True)
         self.assertTrue(self.ax.containers[0][2][0].get_visible())
 
+    def test_no_normalisation_options_on_non_workspace_plot(self):
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py]: Thanks for making that change.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**